src/modules/teachers/components/AssignmentForm.jsx

```diff
@@ -90,14 +90,24 @@ class AssignmentForm extends Component {
     }
     //--------
 
+
+    const target = parseInt(newAssignment.classifications_target);
+    if (newAssignment.subjects.length >= target) {
+      newAssignment.subjects.slice(0, target);
```

Comment:
My bad. This should be `newAssignment.subjects = newAssignment.subjects.slice(0, target);`